src/bam/record.rs

```diff
@@ -136,7 +150,11 @@ impl Record {
         };
 
         let succ = unsafe {
-            htslib::sam_parse1(&mut sam_string, header_view.inner_ptr_mut(), record.inner)
+            htslib::sam_parse1(
+                &mut sam_string,
+                header_view.inner_ptr_mut(),
+                &mut record.inner as *mut htslib::bam1_t,
```

Comment:
For all these pointer conversions, can't we have a helper method?

Reply:
Good point, updated.